src/main/java/cubicchunks/asm/mixin/core/common/MixinWorldServer.java

```diff
@@ -106,7 +108,7 @@
 		for (int cubeX = spawnCubeX - spawnDistance; cubeX <= spawnCubeX + spawnDistance; cubeX++) {
 			for (int cubeZ = spawnCubeZ - spawnDistance; cubeZ <= spawnCubeZ + spawnDistance; cubeZ++) {
 				for (int cubeY = spawnCubeY + spawnDistance; cubeY >= spawnCubeY - spawnDistance; cubeY--) {
-					serverCubeCache.loadCube(cubeX, cubeY, cubeZ, LOAD_OR_GENERATE);
+					serverCubeCache.getCube(new CubeCoords(cubeZ, cubeY, cubeZ), IProviderExtras.Requirement.LIGHT);
```

Comment:
Every cube must have its own CubeCoords instance. The idea is for that instance to be reused everywhere else, thus this should be the only coordinate-object created for the cube. So creating them here should be the opposite of performance wrecking but a small optimization instead.